apps/experiments/tasks.py

```diff
@@ -5,22 +5,32 @@
 from langchain.schema import AIMessage, HumanMessage
 from taskbadger.celery import Task as TaskbadgerTask
 
-from apps.channels.datamodels import BaseMessage
+from apps.channels.datamodels import Attachment, BaseMessage
 from apps.chat.bots import create_conversation
 from apps.chat.channels import WebChannel
 from apps.experiments.models import ExperimentSession, PromptBuilderHistory, SourceMaterial
+from apps.files.models import File
 from apps.service_providers.models import LlmProvider
 from apps.users.models import CustomUser
 from apps.utils.taskbadger import update_taskbadger_data
 
 
 @shared_task(bind=True, base=TaskbadgerTask)
-def get_response_for_webchat_task(self, experiment_session_id: int, message_text: str) -> str:
+def get_response_for_webchat_task(self, experiment_session_id: int, message_text: str, attachments: list) -> str:
     experiment_session = ExperimentSession.objects.get(id=experiment_session_id)
     web_channel = WebChannel(
         experiment_channel=experiment_session.experiment_channel, experiment_session=experiment_session
     )
-    message = BaseMessage(participant_id=experiment_session.participant.identifier, message_text=message_text)
+    message_attachments = []
+    for file_entry in attachments:
+        type, file_id = file_entry.values()
+        message_attachments.append(Attachment(file=File.objects.get(id=file_id), type=type))
```

Comment:
Instead of fetching these here I think it would be better to just pass the file ID and to the fetch just in time e.g. `attachment.get_file()`

Reply:
See https://github.com/dimagi/open-chat-studio/pull/525/commits/ba7e6c81462bdebfb60f6b64d37166a8b17a6a45#diff-6f059913282d6a349d38608d1bc0164c17808f89eb02efb6e0a945e72d2e595f. What I did instead is keep the `file_id` in the attachment and do a bulk fetch in the runnable